Prepare for the Path resolver change to GET

radeusgd opened the pull request:
### Pull Request Description

- Prepares for the Cloud API change, together with a fallback for the old API to avoid problems during migration.
- This PR should be merged before the https://github.com/enso-org/cloud-v2/pull/1236 PR is _deployed_.

### Important Notes

<!--
- Mention important elements of the design.
- Mention any notable changes to APIs.
-->

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

radeusgd committed:
Prepare for the Path resolver change to GET

indiv0 commented on tools/http-test-helper/src/main/java/org/enso/shttp/cloud_mock/PathResolver.java:
```diff
@@ -27,8 +26,19 @@ public boolean canHandle(String subPath) {
+      return;
+    }
+
+    String path = queryString.substring(prefix.length());
```
Don't you still need to URI-decode the path at this point?

radeusgd replied:
Java's `URI::getQuery` returns an already decoded query fragment. `getRawQuery` is undecoded.

radeusgd committed:
update secret tests to use unique names

radeusgd committed:
remove TODO and context

radeusgd committed:
fix from other PR, need it sooner

radeusgd committed:
workaround for https://github.com/enso-org/cloud-v2/issues/1239

radeusgd committed:
temporarily still support the old-style API so that migration is smoother

TODO: revert this commit after https://github.com/enso-org/cloud-v2/pull/1236 is deployed

radeusgd commented on distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Internal/Existing_Enso_Asset.enso:
```diff
@@ -82,12 +84,17 @@ type Existing_Enso_Asset
-            Utils.http_request_as_json HTTP_Method.Post uri payload error_handlers=error_handlers
+        uri = ((URI.from Utils.cloud_root_uri) / "path/resolve") . add_query_argument "path" path
+        r = Utils.http_request_as_json HTTP_Method.Get uri error_handlers=error_handlers
+        ## Workaround to keep compatibility with old cloud API
+           TODO remove it after https://github.com/enso-org/cloud-v2/pull/1236 has been deployed
+        response = r.catch Enso_Cloud_Error error-> case error of
+            Enso_Cloud_Error.Unexpected_Service_Error status _ ->
+                if status.code != 404 then r else
+                    old_uri = (URI.from Utils.cloud_root_uri) / "path/resolve"
+                    old_payload = JS_Object.from_pairs [["path", path]]
+                    Context.Output.with_enabled <| Utils.http_request_as_json HTTP_Method.Post old_uri old_payload error_handlers=error_handlers
+            _ -> r
```
Temporary fallback to make the migration smooth: if the GET endpoint responds with 404, we fall back to the POST endpoint.

radeusgd replied:
The commit d8460d0d063216ec2d53101c7d865da3199f2173 should be reverted after https://github.com/enso-org/cloud-v2/pull/1236 is deployed.

somebody1234 committed:
Fix issue with dependencies?
